pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java

```diff
@@ -607,24 +606,6 @@ static void validateInstancePartitionsTypeMapConfig(TableConfig tableConfig) {
     }
   }
 
-  /**
-   * Detects whether both replicaGroupStrategyConfig and replicaGroupPartitionConfig are set for a given
-   * table. Validation fails because the table would ignore replicaGroupStrategyConfig
-   * when the replicaGroupPartitionConfig is already set.
-   */
-  @VisibleForTesting
-  static void validatePartitionedReplicaGroupInstance(TableConfig tableConfig) {
-    if (tableConfig.getValidationConfig().getReplicaGroupStrategyConfig() == null
-        || MapUtils.isEmpty(tableConfig.getInstanceAssignmentConfigMap())) {
-      return;
-    }
-    for (Map.Entry<String, InstanceAssignmentConfig> entry: tableConfig.getInstanceAssignmentConfigMap().entrySet()) {
-      boolean isNullReplicaGroupPartitionConfig = entry.getValue().getReplicaGroupPartitionConfig() == null;
-      Preconditions.checkState(isNullReplicaGroupPartitionConfig,
```

Comment:
I think we introduced the validation to handle the scenario like what would happen in-case provided `partitionColumn` value is different in `ReplicaGroupStrategyConfig` and `ReplicaGroupPartitionConfig`? 

Looking at the validation logic I can see it's backward incompatible. Can we instead of removing it make sure that if both config has `partitionColumn` value then it should be same otherwise fail?

Reply:
After a second thought, I think we can probably keep it since the validation is applied only when a new table config is uploaded. I found this issue because one table config in the quick start fails this validation, but I can modify the table config to use the new config instead

Reply:
IMO, validation failure is a legitimate backward incompatible  and can possible happen in production system. I think we should fix it. If you want, I can make a change.